src/server/mod.rs

```diff
@@ -43,29 +44,28 @@ macro_rules! try_option(
 impl Server<HttpListener> {
     /// Creates a new server that will handle `HttpStream`s.
     pub fn http(ip: IpAddr, port: Port) -> Server {
+        Server::with_listener(ip, port, HttpListener::Http)
+    }
+}
+
+impl<
+L: NetworkListener<Acceptor=A> + Send,
```

Comment:
Trying to use a where clause here on the impl gave errors that down below `L::Acceptor` couldn't be resolved:

```
/home/sean/code/hyper/src/server/mod.rs:64:101: 64:112 error: associated type `Acceptor` not found for type parameter `L`
/home/sean/code/hyper/src/server/mod.rs:64     pub fn listen_threads<H: Handler>(mut self, handler: H, threads: usize) -> HttpResult<Listening<L::Acceptor>> {
                                                                                                                                               ^~~~~~~~~~~
/home/sean/code/hyper/src/server/mod.rs:138:73: 138:84 error: associated type `Acceptor` not found for type parameter `L`
/home/sean/code/hyper/src/server/mod.rs:138     pub fn listen<H: Handler>(self, handler: H) -> HttpResult<Listening<L::Acceptor>> {
```

This should like be filed as an issue of Rust.